edx_arch_experiments/datadog_diagnostics/middleware.py

```diff
@@ -0,0 +1,75 @@
+"""
+Diagnostic middleware for Datadog.
+"""
+
+import logging
+
+from django.core.exceptions import MiddlewareNotUsed
+from edx_toggles.toggles import WaffleFlag
+
+log = logging.getLogger(__name__)
+
+# .. toggle_name: datadog.diagnostics.log_root_span
+# .. toggle_implementation: WaffleFlag
+# .. toggle_default: False
+# .. toggle_description: Enables logging of Dataodog root span IDs for web requests.
+# .. toggle_warning: This is a noisy feature and so it should only be enabled
+#   for a percentage of requests.
```

Comment:
Right now we are dealing with both Splunk and Datadog, so this makes sense. Also, our Log Ingest was high (over-budget), but SRE may have taken care of that for us? Otherwise, in the future, we could instead control this via Log Index rules that could be tuned based on noisiness. Although the log messages might still affect disk usage, if it really is high.